pkg/sql/logictest/tests/local/local_test: TestLogic_guardrails failed

pkg/sql/logictest/tests/local/local_test.TestLogic_guardrails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10415101?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10415101?buildTab=artifacts#/) on release-23.1 @ [a9a6eb0782dbce44639aa629a22e979da87c24ba](https://github.com/cockroachdb/cockroach/commits/a9a6eb0782dbce44639aa629a22e979da87c24ba):


```
=== RUN   TestLogic_guardrails
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/71711051c96cfc11cc9634f70d9a3b22/logTestLogic_guardrails1988091619
    test_log_scope.go:79: use -show-logs to present logs inline
[13:01:05] setting distsql_workmem='91854B';
    logic.go:2886: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2778/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:30: SELECT * FROM guardrails
        expected:
        txn has read 51 rows, which is above the limit
        got:
        pq: txn has read 56 rows, which is above the limit: TxnID f4496f6e-7f1b-4601-807c-f55384dc8a95 SessionID 1766136c192b7ef00000000000000001
[13:01:05] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2778/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails with config local: 9 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2778/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:33: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2778/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:33: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/71711051c96cfc11cc9634f70d9a3b22/logTestLogic_guardrails1988091619
--- FAIL: TestLogic_guardrails (1.30s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_guardrails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-28513

--- a/pkg/sql/logictest/testdata/logic_test/guardrails
+++ b/pkg/sql/logictest/testdata/logic_test/guardrails
@@ -1,3 +1,7 @@
+# LogicTest: !metamorphic-batch-sizes
+# We disable metamorphic batch sizes so that we read a predictable number of
+# rows in each limited scan.
+
 statement ok
 CREATE TABLE guardrails (i INT PRIMARY KEY);
 INSERT INTO guardrails SELECT generate_series(1, 100)

